framework/TApplicationComponent.php

```diff
@@ -43,6 +46,40 @@ public function getAutoGlobalListen()
 		return true;
 	}
 	
+	/**
+	 * This caches the 'fx' events for PRADO classes in the global state
+	 * @param object $class
+	 * @return string[] fx events from a specific class
+	 */
+	protected function getClassFxEvents($class)
+	{
```

Comment:
Wow, I see a lot of optimizations going on here.
The method is quite complex, but i didn't find any case where it fails.
I suppose it took a lot to get it so well done.

Reply:
It has been tested and tested and tested, yes.  The final touch was to run on Normal and Performance Mode only.  I figured that there needed to be a way to speed this up more.